### PR TITLE
Migrate logging, metrics & management packages to new fabric8-maven-plugin.

### PR DESCRIPTION
**Pull request by jimmidyson**
This includes upgrading Prometheus, grafana, kibana, elasticsearch & fluentd. And also adds node exporter daemonset to get loads more metrics from nodes.

### CODE REVIEW

**Comment by jimmidyson** [comment]
[merge]

**Comment by jimmidyson** [comment]
[merge]

File: elasticsearch/pom.xml [hunk 1]
```diff
@@ -70,25 +47,33 @@
...
+          <execution>
+            <id>resource</id>
+            <goals>
+              <goal>resource</goal>
+              <goal>build</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>json</id>
+            <phase>none</phase>
```
Comment (rhuss) [inline elasticsearch/pom.xml]: just curious: Why is this required ?

Comment (rhuss) [inline elasticsearch/pom.xml]: maybe because of bindings in the parent pom ?

Comment (jimmidyson) [inline elasticsearch/pom.xml]: :white_check_mark: 

Not sure if this will release properly tbh - do I need to do some extra do you think

Comment (rhuss) [inline elasticsearch/pom.xml]: For `<configurations>` one could use `combine.self="override"` to override everything, but this is not available for `<executions>`. Unfortunately this is indeed a bit tricky because of this mixed setup.

The only alternatives to `<phase>none</phase>` I can think of are:

* Dont bind executions in the parent pom.xml at all but always where it is used (e.g. where f-m-p-2 is still used).
* Use different profiles, so that you could both binding IN the parent pom.xml. These profiles could be activated in the child poms by setting a property (e.g. `use.fmp.3`) and a property activation for the profiles. This is probably the best solution.

Comment (rhuss) [inline elasticsearch/pom.xml]: Maybe sth like

```xml
<profiles>
  <profile>
    <id>f-m-p-3</id>
    <activation>
      <property>
        <name>fabric8-maven-plugin.majorVersion</name>
        <value>3</value>
      </property>
    </activation>
    ...
  </profile>
</profiles>
```

in the parent pom, with two profiles for each variant. An then set in each individual project pom.xml `fabric8-maven-plugin.majorVersion` to `2` or `3`.

File: elasticsearch/src/main/fabric8/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          tcpSocket:
+            port: 9300
+          initialDelaySeconds: 30
+        readinessProbe:
+          httpGet:
+            port: 9200
+            path: /_template/logstash
+          initialDelaySeconds: 5
+      - name: logstash-template
+        image: ${fabric8.dockerUser}elasticsearch-logstash-template:${project.version}
```
Comment (rhuss) [inline elasticsearch/src/main/fabric8/deployment.yml]: FYI: I'm going to make the registry more transparent, so that we can avoid this nasty `${fabric8.dockerUser}` hack. This will probably done by an `RegistryEnricher` which post-processes the generated files and then, according to the given configuration, adds a registry and/or user.

Comment (jstrachan) [inline elasticsearch/src/main/fabric8/deployment.yml]: we also need to be able to swizzle when staging multiple releases to use a staging registry versus the real registry (e.g. testing / staging / production registries). Docker registries are unfortunately the one thing we can't use kubernetes services for :(

I did some code to swizzle registries in docker images in kubernetes resources somewhere; though can't remember where ;) @rawlingsj can you remember by any chance? Struggling to remember where that code went ;)
